distribution/lib/Standard/Base/0.0.0-dev/src/Error/Extensions.enso

```diff
@@ -125,14 +125,31 @@ Error.map_error f = this.catch (x -> Error.throw (f x))
 ## ADVANCED
    UNSTABLE
 
-   Returns the attached stack trace of the error. The ordering of the resulting
-   vector is such that the top stack frame is the first element.
-Error.get_stack_trace : Vector.Vector Stack_Trace_Element
-Error.get_stack_trace =
-    prim_stack = this.primitive_get_stack_trace
+   Returns the attached stack trace of the given throwable. Can be used to get
+   an Enso friendly stack trace from native Java exceptions.
+
+   The ordering of the resulting vector is such that the top stack frame is the
+   first element.
+Panic.get_attached_stack_trace : Caught_Panic | Throwable -> Vector.Vector Stack_Trace_Element
+Panic.get_attached_stack_trace error =
+    throwable = case error of
+        Caught_Panic _ internal_original_exception -> internal_original_exception
+        throwable -> throwable
```

Comment:
Why does this exist? If I'm reading your code correctly, there's no way to interact with panics other than through `CaughtPanic`?

Reply:
1. to be able to get the stacktrace of a raw Java exception, if you get it from somewhere.
2. Because it is exactly used as an internal helper to make `Caught_Panic` able to have the `stack_trace` method.

Reply:
1. possibly from `caught_panic.payload.cause` (I know there we simply have `caught_panic.stack_trace` but the `payload` may be passed further down) or from `catch_java_exception` unless we decide to remove this function (but it's not clear yet). I just feel like that function may have its uses somewhere.

Reply:
OK, I'll believe you